src/charm.py

```diff
@@ -0,0 +1,1017 @@
+        )
+        self.framework.observe(
+            self.on.update_oauth_client_action, self._on_update_oauth_client_action
+        )
+        self.framework.observe(
+            self.on.delete_oauth_client_action, self._on_delete_oauth_client_action
+        )
+        self.framework.observe(
+            self.on.list_oauth_clients_action, self._on_list_oauth_clients_action
+        )
+        self.framework.observe(
+            self.on.revoke_oauth_client_access_tokens_action,
+            self._on_revoke_oauth_client_access_tokens_action,
+        )
+        self.framework.observe(self.on.rotate_key_action, self._on_rotate_key_action)
+
+        self.framework.observe(
+            self.loki_consumer.on.promtail_digest_error,
+            self._promtail_error,
+        )
+
+    @property
+    def _hydra_service_params(self) -> str:
+        ret = ["--config", str(self._hydra_config_path)]
+        if self.config["dev"]:
+            logger.warning("Running Hydra in dev mode, don't do this in production")
+            ret.append("--dev")
+
+        return " ".join(ret)
+
+    @property
+    def _hydra_layer(self) -> Layer:
+        """Returns a pre-configured Pebble layer."""
+        layer_config = {
+            "summary": "hydra-operator layer",
+            "description": "pebble config layer for hydra-operator",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "entrypoint of the hydra-operator image",
+                    "command": '/bin/sh -c "{} {} 2>&1 | tee -a {}"'.format(
+                        self._hydra_service_command,
+                        self._hydra_service_params,
+                        str(self._log_path),
+                    ),
+                    "startup": "disabled",
+                }
+            },
+            "checks": {
+                "version": {
+                    "override": "replace",
+                    "exec": {"command": "hydra version"},
+                },
+                "ready": {
+                    "override": "replace",
+                    "http": {"url": f"http://localhost:{HYDRA_ADMIN_PORT}/health/ready"},
+                },
+            },
+        }
+
+        if self._tracing_ready:
+            layer_config["services"][self._container_name]["environment"] = {
+                "TRACING_PROVIDER": "otel",
+                "TRACING_PROVIDERS_OTLP_SERVER_URL": self._get_tracing_endpoint_info(),
+                "TRACING_PROVIDERS_OTLP_INSECURE": "true",
+                "TRACING_PROVIDERS_OTLP_SAMPLING_SAMPLING_RATIO": "1.0",
+            }
+
+        return Layer(layer_config)
+
+    @property
+    def _hydra_service_is_created(self) -> bool:
+        try:
+            self._container.get_service(self._container_name)
+        except (ModelError, RuntimeError):
+            return False
+        return True
```

Comment:
```suggestion
        return self._container.can_connect() and self._container_name  in self._container.get_services()
```
So we don't need to capture the `RuntimeError` which may caused by some unknown problems.